users/service.go

```diff
@@ -45,8 +68,22 @@ type Service interface {
 	// other reason, non-nil error values are returned in response.
 	Identify(string) (string, error)
 
-	// Get authenticated user info for the given token.
+	// Get authenticated user info for the given token
 	UserInfo(ctx context.Context, token string) (User, error)
+
+	// GenerateResetToken email where mail will be sent.
+	// host is used for generating reset link.
+	GenerateResetToken(_ context.Context, email, host string) error
+
+	// ChangePassword change users password for authenticated user.
+	ChangePassword(_ context.Context, authToken, password, oldPassword string) error
```

Comment:
Consider renaming to `UpdatePassword` (`UpdateThing`, `UpdateChannel`).

Reply:
I've considered it and somehow this is best for since UpdatePassword is in repo and it is like primitive action that is part of ChangePassword and ResetPassword